mypy/main.py

```diff
@@ -205,6 +207,39 @@ def invert_flag_name(flag: str) -> str:
     return '--no-{}'.format(flag[2:])
 
 
+if sys.platform == 'win32':
+    python_executable_prefix = 'py -{}'
+else:
+    python_executable_prefix = 'python{}'
+
+
+def _python_version_from_executable(python_executable: str) -> Tuple[int, int]:
+    try:
+        check = subprocess.check_output([python_executable, '-c',
+                                         'import sys; print(repr(sys.version_info[:2]))'],
+                                        stderr=subprocess.STDOUT).decode()
+    except subprocess.CalledProcessError:
+        return sys.version_info[:2]
+    else:
+        return ast.literal_eval(check)
+
+
+def _python_executable_from_version(python_version: Tuple[int, int]) -> Optional[str]:
+    str_ver = '.'.join(map(str, python_version))
+    try:
+        sys_exe = subprocess.check_output(python_executable_prefix.format(str_ver).split(' ') +
```

Comment:
Rather than splitting a string, perhaps:

```
if sys.platform == 'win32':
    python_executable_guess = lambda v: ['py', '-{}'.format(v)]
else:
    python_executable_guess = lambda v: ['py{}'.format(v)]
```